improve example documentation in Annex (#99)
admin-shell-io/aas-specs-security: documentation/IDTA-01004/modules/ROOT/pages/annex/text-access-rule-examples.adoc

[[allow-read-submodels-id-pattern]]
[[example-time-based-submodel-id-pattern]]
== EXAMPLE 9: Example with access constraints 

For users authenticated via "companyName", "companyName" shall be "company1-name":

Allow READ of Submodels with semanticId "SemanticID-Nameplate" or "SemanticID-TechnicalData" if the Submodel/id starts with "https://company1.com/" but only at working days (Monday to Friday) between 9:00 and 17:00 Utc.

[source,bnf,linenums]
----
include::partial$examples/allow-read-submodels-id-pattern.bnf[]
----

== EXAMPLE 10: Example with allowing creation within Submodel

For users authenticated with role "person with legitimate interest":

> Allow CREATE of SubmodelElements within Submodels with semanticId "CertificateSet".

Reviewer: In the FORMULA, $sme#semanticId must equal "CertificateSet". Therefore, any Submodel can be created, provided that it contains only SubmodelElements with the semanticId "CertificateSet". Likewise, only SubmodelElements with the semanticId "CertificateSet" can be created within any Submodel.

```suggestion
Allow CREATE operations for Submodels containing only SubmodelElements with the semanticId "CertificateSet", as well as for SubmodelElements with the semanticId "CertificateSet".
```

[source,bnf,linenums]
----
include::partial$examples/allow-create-only-specific.bnf[]
----

[[example-filter-statement]]
== EXAMPLE 11: Example with FILTER statement

For users authenticated via "BusinessPartnerNumber", "BusinessPartnerNumber" shall be "BPNL00000000000A":

Allow READ of AssetAdministrationShellDescriptors containing the following specificAssetIds:

* there exists a specificAssetId with name "manufacturerPartId" and  value "99991" and  externalSubjectId  "PUBLIC_READABLE"
* there additionally exists a specificAssetId with name "customerPartId" and value "ACME001"

Not all specificAssetIds of the AssetAdministrationShellDescriptors fulfilling theses constraints are returned but only 

* the two specificAssetIds above used for selecting the AssetAdministrationShellDescriptors
* and additionally all those specificAssetIds that are public, i.e. with externalSubjectId equal to "PUBLIC_READABLE"

> * and all specificAssetIds related to the autenticated user, i.e. those with externalSubjectId equal to the BusinessPartnerNumber" of the user

Author: ```suggestion
* and all specificAssetIds related to the authenticated user, i.e. those with externalSubjectId equal to the BusinessPartnerNumber" of the user
```

> * and all specificAssetIds with name "partInstanceid"

Author: ```suggestion
* and all specificAssetIds with name "partInstanceId"
```
